Simplify authoring guidelines for "Reference Solution" snippet

hobovsky opened the pull request:
After asking FArekkusu for opinion, he pointed out that guidelines related to "reference solution" snippet contain some unnecessary remarks (see #177). He mostly confirmed what I felt while writing the guidelines, and I think that for some reason I just wanted to add some meaning to this snippet, which might be not _that_ meaningful after all. Most of requirements I put on it should be handled by other parts, like clear, complete description, and comprehensive test suite.

Tell me what you think and if simplified version of the remarks, limited only to "keep it maintainable" is OK, or you have another idea.

Discussion: #177 
Deploy preview: [here](https://deploy-preview-179--reverent-edison-2864ea.netlify.app/authoring/guidelines/reference-solution/)

DonaldKellett commented on content/authoring/guidelines/reference-solution.md:
```diff
@@ -7,15 +7,8 @@ next: /authoring/guidelines/translation/
-- **Conform to [General Coding Guidelines][authoring-guidelines-general-coding]**{id="general"}: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators. The reference solution is not for you to write, but fo ...
-- When creating a new kata, especially a performance oriented one, the reference solution should reflect **the minimal solution intended by the kata author to pass**{id="minimal-ref"}. Sometimes maintainers and translators are not sure what the author's original intention behind the kata was in terms of its difficulty, edge cases and additional restrictions. The reference solution of the initial  ...
-- For reasons mentioned above, when a kata is translated, **the meaning of the reference solution from the initial version should be preserved**{id="keep-reqs"}. It's especially important for performance oriented kata. The reference solution of a translation should not significantly stand out from the original one. They do not have to be totally equivalent, and they can use different techniques,  ...
+The reference solution snippet, created by the kata author or translator, is a snippet which might appear as not very relevant, but when prepared incorrectly, it still can incur some additional, unnecessary maintenance cost. It might appear that reference solution is just a solution like every other submitted by a user, and the only important thing is to pass the tests. However, it's something mo ...
```
```suggestion
The reference solution snippet, created by the kata author or translator, is a snippet which might appear as not very relevant, but when prepared incorrectly, it still can incur some additional, unnecessary maintenance cost. It might appear that the reference solution is just a solution like every other submitted by a user, and that the only important thing is to pass the tests. However, it's something more: the reference solution is run every time a kata or translation is republished, and it should be prepared in a way which would prevent the publication from failing in an unexpected way.
```

DonaldKellett commented on content/authoring/guidelines/reference-solution.md:
```diff
@@ -7,15 +7,8 @@ next: /authoring/guidelines/translation/
-- For reasons mentioned above, when a kata is translated, **the meaning of the reference solution from the initial version should be preserved**{id="keep-reqs"}. It's especially important for performance oriented kata. The reference solution of a translation should not significantly stand out from the original one. They do not have to be totally equivalent, and they can use different techniques,  ...
+The reference solution snippet, created by the kata author or translator, is a snippet which might appear as not very relevant, but when prepared incorrectly, it still can incur some additional, unnecessary maintenance cost. It might appear that reference solution is just a solution like every other submitted by a user, and the only important thing is to pass the tests. However, it's something mo ...
 
+For above reasons, this snippet **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]**: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators.
```
```suggestion
For the above reasons, this snippet **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]**: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators.
```

kazk commented on content/authoring/guidelines/reference-solution.md:
```diff
@@ -7,15 +7,8 @@ next: /authoring/guidelines/translation/
-- **Conform to [General Coding Guidelines][authoring-guidelines-general-coding]**{id="general"}: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators. The reference solution is not for you to write, but fo ...
-- When creating a new kata, especially a performance oriented one, the reference solution should reflect **the minimal solution intended by the kata author to pass**{id="minimal-ref"}. Sometimes maintainers and translators are not sure what the author's original intention behind the kata was in terms of its difficulty, edge cases and additional restrictions. The reference solution of the initial  ...
-- For reasons mentioned above, when a kata is translated, **the meaning of the reference solution from the initial version should be preserved**{id="keep-reqs"}. It's especially important for performance oriented kata. The reference solution of a translation should not significantly stand out from the original one. They do not have to be totally equivalent, and they can use different techniques,  ...
+The reference solution snippet, created by the kata author or translator, is a snippet which might appear as not very relevant, but when prepared incorrectly, it still can incur some additional, unnecessary maintenance cost. It might appear that the reference solution is just a solution like every other submitted by a user, and that the only important thing is to pass the tests. However, it's som ...
 
+For the above reasons, this snippet **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]**: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators.
```
Isn't this too wordy? It's difficult to understand because of that and the passive tone.

```suggestion
The reference solution **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]** because it must be maintained, unlike other submitted solutions. Always think about the future maintainers and translators.
```

I'm not sure having a separate page for this is worth it, but I do hate fixing **inelegant** one-liners that the translator was once proud of before disappearing.

Blind4Basics replied:
(note: the smiley is about the comment below the suggestion)

What is the "reference solution" should be told first, I believe. Currently, this cannot be deduced from the text. For instance, when I hear "ref solutino", I always think about the one used for the random tests, in the tests suite.
Using kazk suggestion:

```suggestion
The reference solution is the solution of the author/translator that is showing up in the kata solutions page. It **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]** because it must be maintained if changes are done to the tests, unlike other submitted solutions. Always think about the future maintainers and translators.
```

hobovsky replied:
I wanted users to realize that this snippet poses a maintenance cost, and not that it's just "write once and forget". I have an impression that users do not realize that sloppiness here can cause some unnecessary struggle to maintainers. But indeed I could be too wordy, because I often am :)

hobovsky commented:
I removed the wordy paragraph in the beginning, and added a remark on disambiguation between "reference solution" snippet and a code used to generate expected answers. I think it's still worth to have a separate page for this part.

You guys think it's better now?

I also raised #180 on styling of the note.
